### PR TITLE
Fix external symbols related to keys ec, ecx, sm2, dh & dsa

### PR DESCRIPTION
**Pull request by slontis**
Partial fix for #12964

This adds ossl_ names for the following symbols:

````
ec_*, ecx_*, ecdh_*, ecdsa_*, sm2_*
dsa_check_pairwise, dsa_check_params, dsa_check_priv_key, dsa_check_pub_key, dsa_check_pub_key_partial,
dsa_do_sign_int, dsa_ffc_params_fromdata,
dsa_generate_ffc_parameters, dsa_generate_public_key,
dsa_get0_params, dsa_key_fromdata, dsa_new_with_ctx, dsa_pkey_method, dsa_sign_int
dh_new_by_nid_ex, dh_new_ex, dh_generate_ffc_parameters, dh_generate_public_key,
dh_get_named_group_uid_from_size, dh_gen_type_id2name, dh_gen_type_name2id,
dh_cache_named_group, dh_get0_params, dh_get0_nid,
dh_params_fromdata, dh_key_fromdata, dh_params_todata, dh_key_todata,
dh_check_pub_key_partial, dh_check_priv_key, dh_check_pairwise,
dh_get_method, dh_buf2key, dh_key2buf, dh_KDF_X9_42_asn1,
dh_pkey_method, dhx_pkey_method
````

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
EC got a bit large due to the number of existing ec functions that were already in 1.1.1
It might be easier to review by commits

File: crypto/ec/ecp_nistputil.c [hunk 1]
```diff
@@ -49,7 +49,8 @@
  * of size 'felem_size'. tmp_felems needs to point to a temporary array of
  * 'num'+1 field elements for storage of intermediate values.
  */
-void ec_GFp_nistp_points_make_affine_internal(size_t num, void *point_array,
+void
+ossl_ec_GFp_nistp_points_make_affine_internal(size_t num, void *point_array,
```
Comment (paulidale) [inline crypto/ec/ecp_nistputil.c:53]: I'd have broken the line at the `(` not the space after void.  It makes the argument list one or two lines.  Not critical.

File: include/crypto/ecx.h [hunk 2]
```diff
@@ -76,13 +76,12 @@ struct ecx_key_st {
 
 typedef struct ecx_key_st ECX_KEY;
 
-size_t ecx_key_length(ECX_KEY_TYPE type);
```
Comment (paulidale) [inline include/crypto/ecx.h:79]: Was this meant to be removed?

**Review by paulidale — APPROVED** [review]
Approved, unless the answer to one of the points raised is controversial.

File: include/crypto/ecx.h (hunk 2 above)
Comment (slontis) [reply include/crypto/ecx.h:79]: yes, it looks like a leftover - it doesn't exist anywhere.

File: crypto/ec/ecp_nistputil.c (hunk 1 above)
Comment (slontis) [reply crypto/ec/ecp_nistputil.c:53]: richard does not like the breaking at the (..

**Comment by slontis** [comment]
Had to rebase.. Changes are in the last 2 fixups to fix recently added internal symbols for ec and dh.

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Had to rebase due to pkey ctrl changes.. Last commit fixes the new files.

**Comment by slontis** [comment]
@paulidale can you reapprove please

File: crypto/evp/pmeth_lib.c [hunk 3]
```diff
@@ -1326,6 +1326,7 @@ int EVP_PKEY_CTX_ctrl_uint64(EVP_PKEY_CTX *ctx, int keytype, int optype,
     return EVP_PKEY_CTX_ctrl(ctx, keytype, optype, cmd, 0, &value);
 }
 
+
```
Comment (paulidale) [inline crypto/evp/pmeth_lib.c:1329]: Two blank lines??

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Rebasing again - should not need reapproval.. I think I will just merge this soon as this is causing many rebases..

**Comment by paulidale** [comment]
Go for it.

**Comment by slontis** [comment]
Thanks for reviewing. Merged to master.

**Pull request closed without merging**